funding/manager.go

```diff
@@ -2365,11 +2437,30 @@ func (f *Manager) handleFundingSigned(peer lnpeer.Peer,
 	pendingChanID, ok := f.signedReservations[msg.ChanID]
 	delete(f.signedReservations, msg.ChanID)
 	f.resMtx.Unlock()
+
+	// Create the channel identifier and set the channel ID.
+	//
+	// NOTE: we may get an empty pending channel ID here if the key cannot
+	// be found, which means when we cancel the reservation context in
+	// `failFundingFlow`, we will get an error. We will also send an error
+	// msg to our peer with this empty pending channel ID, which she/he
+	// won't be able to understand.
+	//
+	// TODO(yy): refactor the funding flow to fix this case.
+	cid := newChanIdentifier(pendingChanID)
+	cid.setChanID(msg.ChanID)
+
+	// If the pending channel ID is not found, fail the funding flow.
 	if !ok {
+		// NOTE: we directly overwrite the pending channel ID here for
+		// this rare case, which can be removed once we can make sure
+		// the pending channel ID is always found here.
```

Comment:
We *could* check the channel ID in `peer/brontide.go` and handle it there if there's no pending channel for that ID.  Then we could guarantee that only real channel IDs end up in `handleFundingSigned`.  But please don't do that -- funding manager is a better place for this.

I think we should replace the second part of the comment:
```suggestion
		// this rare case since we don't have a valid pending channel ID.
```

Reply:
yeah that's not what brontide is for. Updated the comments.